core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1330,7 +1330,7 @@ private Optional<RemoteLogSegmentMetadata> findNextSegmentMetadata(RemoteLogSegm
                 : Optional.empty();
     }
 
-    private RecordBatch findFirstBatch(RemoteLogInputStream remoteLogInputStream, long offset) throws IOException {
+    protected RecordBatch findFirstBatch(RemoteLogInputStream remoteLogInputStream, long offset) throws IOException {
```

Comment:
default access? Also, in Kafka code base we add a `// Visible for testing` comment whenever we change access due to tests